lib/iris/fileformats/grib/message.py

```diff
@@ -124,6 +160,15 @@ def data(self):
             raise TranslationError(fmt.format(template))
         return data
 
+    def __getstate__(self):
```

Comment:
I'm not convinced we need this, or for that matter any of "tests.integration.test_pickle.TestGribMessage".
As far as I can see, messages don't form any part of a cube :
only the ```fileformats.grib.message._DataProxy``` and ```._MessageLocation```do that
 * If this is ***not*** needed it should really go (and probably that pickle test too).
 * or, if it ***is*** needed then this doesn't necessarily solve the problem, as an unpickled message could not access any keys not already read in.

@rhattersley can you shed light on why we though we needed to **pickle** grib messages,
and could this maybe now be obsolete ??